frappe/database/database.py

```diff
@@ -79,26 +79,19 @@ class InvalidColumnName(frappe.ValidationError):
 
 	def __init__(
 		self,
-		host=None,
-		user=None,
-		password=None,
-		ac_name=None,
-		use_default=0,
-		port=None,
+		host,
+		user,
+		password,
+		port,
+		cur_db_name=None,  # not sensible on mariadb root connection
 	):
 		self.setup_type_map()
-		self.host = host or frappe.conf.db_host or "127.0.0.1"
-		self.port = port or frappe.conf.db_port or ""
-		self.user = user or frappe.conf.db_name
-		self.db_name = frappe.conf.db_name
+		self.host = host
+		self.port = port
+		self.user = user
+		self.cur_db_name = cur_db_name
```

Comment:
The core of this PR: do not (quite implicitly) default on the database interface. This change requires being explicit at the calling site.